pkg/apiserver/watch.go

```diff
@@ -32,19 +34,48 @@ import (
 	"golang.org/x/net/websocket"
 )
 
-var connectionUpgradeRegex = regexp.MustCompile("(^|.*,\\s*)upgrade($|\\s*,)")
+var (
+	connectionUpgradeRegex = regexp.MustCompile("(^|.*,\\s*)upgrade($|\\s*,)")
+
+	// nothing will ever be sent down this channel
+	neverExitWatch <-chan time.Time = make(chan time.Time)
+)
 
 func isWebsocketRequest(req *http.Request) bool {
 	return connectionUpgradeRegex.MatchString(strings.ToLower(req.Header.Get("Connection"))) && strings.ToLower(req.Header.Get("Upgrade")) == "websocket"
 }
 
+// TimeoutMonitor abstracts watch timeout logic for testing
+type TimeoutMonitor interface {
```

Comment:
This is a factory, right? It doesn't monitor anything, right? ```timeoutFactory``` would be much clearer name.